mermithid/processors/misc/TritiumAndEfficiencyBinner.py

```diff
@@ -0,0 +1,136 @@
+function.
+Author: A. Ziegler, E. Novitski, C. Claessens
+Date:3/4/2020
+
+This takes efficiency informations and interpolates between frequency points.
+Then, you dump in tritium data.
+It assigns an efficiency and efficiency uncertainty upper and lower bounds to each event.
+It also bins events and defines an efficiency and efficiency uncertainty upper and
+lower bound by integrating the interpolated efficiency over each bin.
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+import scipy
+import sys
+import json
+from morpho.utilities import morphologging, reader
+from morpho.processors import BaseProcessor
+from mermithid.misc import Constants
+
+logger = morphologging.getLogger(__name__)
+
+
+
+__all__ = []
+__all__.append(__name__)
+
+def Energy(f, B=None, Theta=None):
+    #print(type(F))
+    if B==None:
+        B = 0.95777194923080811
+    emass_kg = Constants.m_electron()*Constants.e()/(Constants.c()**2)
+    if isinstance(f, list):
+        gamma = [(Constants.e()*B)/(2.0*TMath.Pi()*emass_kg) * 1/(F) for F in f]
+        return [(g -1)*Constants.m_electron() for g in gamma]
+    else:
+        gamma = (Constants.e()*B)/(2.0*TMath.Pi()*emass_kg) * 1/(f)
+        return (gamma -1)*Constants.m_electron()
+
+class TritiumAndEfficiencyBinner(BaseProcessor):
+
+    def InternalConfigure(self, params):
+        '''
+        Configure
+        '''
+
+        # Read other parameters
+        self.namedata = reader.read_param(params, 'variables', "required")
+        self.N = reader.read_param(params, 'N', 'N')
+        self.eff_eqn = reader.read_param(params, 'efficiency', '1')
+        self.bins = reader.read_param(params, 'bins', [])
+        self.asInteger = reader.read_param(params, 'asInteger', False)
+        self.energy_or_frequency = reader.read_param(params, 'energy_or_frequency', 'energy')
+        self.efficiency_filepath = reader.read_param(params, 'efficiency_filepath', '/host/combined_energy_corrected_eff_at_quad_trap_frequencies.json')
+        self.fss_bins = reader.read_param(params, "fss_bins", False)
+        # If self.fss_bins is True, self.bins is ignored and overwritten
+
+        # initialize the histogram to store the corrected data
+        if self.energy_or_frequency == 'energy':
+            print(sys.getrefcount(self.corrected_data))
+            self.output_bin_variable='KE'
+        elif self.energy_or_frequency == 'frequency':
+            self.output_bin_variable='F'
+
+        if self.fss_bins == True:
+            a = self.GetEfficiencyFileContent()
+            self.bin_centers = a['frequencies']
+            self.bins = np.array(self.bin_centers) - (self.bin_centers[1]-self.bin_centers[0])/2
+            self.bins = np.append(self.bins, [self.bin_centers[-1]+(self.bin_centers[1]-self.bin_centers[0])/2])
+        else:
+            self.bin_centers = self.bins[0:-1]+0.5*(self.bins[1]-self.bins[0])
+        return True
+
+    def InternalRun(self):
+        print('namedata:',self.namedata)
```

Comment:
You can use the logger to make formatted prints on screen; it is then easier to find where this line was printed from...